PostFreeSurfer/scripts/GenerateStructuralScenes.sh

```diff
@@ -0,0 +1,327 @@
+   echo "... relative path to StudyFolder (from OutputSceneFolder): $relPathToStudy"
+fi
+
+# Define some convenience variables
+AtlasSpaceFolder="$StudyFolder/$Subject/MNINonLinear"
+mesh="164k_fs_LR"
+
+if [[ -d "$AtlasSpaceFolder/xfms" ]]; then
+    if ((verbose)); then
+        echo "Subject folder appears to be okay."
+    fi
+else
+    log_Err_Abort "ERROR:  Subject folder missing expected directory MNINonLinear/xfms"
+fi
+
+# Replace dummy strings in the template scenes to generate
+# a scene file appropriate for each subject
+SubjectSceneFile="$OutputSceneFolder/$Subject.structuralQC.wb_scene"
+sed -e "s|${StudyFolderDummyStr}|${relPathToStudy}|g" \
+    -e "s|${SubjectIDDummyStr}|${Subject}|g" \
+    -e "s|${TemplatesFolderDummyStr}|${relPathToTemplates}|g" \
+    "$TemplatesFolder"/TEMPLATE_structuralQC.scene > "$SubjectSceneFile"
+
+# If StrainJ maps don't exist for the various registrations, 
+# but ArealDistortion maps do, use those instead
+for regName in FS MSMSulc MSMAll; do
+    if [[ ! -e "$AtlasSpaceFolder/$Subject.StrainJ_$regName.$mesh.dscalar.nii" && -e "$AtlasSpaceFolder/$Subject.ArealDistortion_$regName.$mesh.dscalar.nii" ]]; then
+        echo "... using ArealDistortion_${regName} map in place of StrainJ_${regName}"
+        # Following version of sed "in-place" replacement should work on both Linux and MacOS
+        sed -i.bak "s|StrainJ_${regName}|ArealDistortion_${regName}|g" "$SubjectSceneFile"
+        rm -f "$SubjectSceneFile.bak"
+    fi
+done
+
+## Map the T1w_acpc space volume into MNI152 space, using just the affine (linear) component
+## [Similar to the 'MNINonLinear/xfms/T1w_acpc_dc_restore_brain_to_MNILinear.nii.gz' volume
+## (created in AtlasRegistrationToMNI152_FLIRTandFNIRT.sh) 
+## except applied to the NON-brain-extracted volume].
+acpc2MNILinear="$AtlasSpaceFolder/xfms/acpc2MNILinear.mat"
+if [[ -e "$acpc2MNILinear" ]]; then
+    nativeVol=T1w_acpc_dc_restore
+    volumeIn="$AtlasSpaceFolder/../T1w/$nativeVol.nii.gz"
+    volumeRef="$AtlasSpaceFolder/T1w_restore.nii.gz"
+    volumeOut="$OutputSceneFolder/$Subject.${nativeVol}_to_MNILinear.nii.gz"
+    if [[ "$maskDilate" != "-1" ]]
+    then
+        tempresample=$(tempfiles_create strucQC_resamp_XXXXXX.nii.gz)
+        wb_command -volume-affine-resample \
+            "$volumeIn" "$acpc2MNILinear" "$volumeRef" CUBIC "$tempresample" \
+            -flirt "$volumeIn" "$volumeRef"
+        #noise in the background makes this file bigger than necessary, mask it out
+        #use the existing MNINonLinear brain mask, close enough with large dilation
+        tempmask=$(tempfiles_create strucQC_mask_XXXXXX.nii.gz)
+        wb_command -volume-dilate \
+            "$AtlasSpaceFolder/brainmask_fs.nii.gz" \
+            "$maskDilate" \
+            NEAREST \
+            "$tempmask"
+        wb_command -volume-math 'data * (mask > 0)' "$volumeOut" \
+            -var data "$tempresample" \
+            -var mask "$tempmask"
```

Comment:
Should we explicitly remove the temp files that we've created, to leave the temp space as we found it?

Reply:
I think that is good practice.

Reply:
tempfiles_create does a tempfiles_add internally, which does `rm` in a `trap ... EXIT`, so this is already taken care of.  This is the purpose of `tempfiles.shlib`, temporaries it knows about get deleted without needing to remember to `rm` them manually.